src/agent/agent_instance.cpp

```diff
@@ -40,7 +40,7 @@
 
 namespace otbr {
 
-AgentInstance::AgentInstance(Ncp::Controller *aNcp)
+AgentInstance::AgentInstance(otbr::Ncp::ControllerOpenThread &aNcp)
```

Comment:
Is it necessary to be fully-qualified? We are already in `otbr` namespace here.
```suggestion
AgentInstance::AgentInstance(Ncp::ControllerOpenThread &aNcp)
```